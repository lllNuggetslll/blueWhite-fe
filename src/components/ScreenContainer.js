import React from "react";
import { node } from "prop-types";
import styled from "styled-components";
import TopBar from "./TopBar";

const Container = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  height: 100%;
  background: rgb(2, 0, 36);
  background: linear-gradient(
    90deg,
    rgba(2, 0, 36, 1) 0%,
    rgba(9, 9, 121, 1) 17%,
    rgba(0, 212, 255, 1) 100%
  );
`;

const ChildContainer = styled.div`
  display: flex;
  flex-direction: column;
  text-align: center;
  padding: 60px;
  border-radius: 10px;
  background: white;
  box-shadow: -5px 2px 26px 0px rgba(0, 0, 0, 0.44);
`;

const ScreenContainer = ({ children }) => {
  return (
    <Container>
      <TopBar />
      <ChildContainer>{children}</ChildContainer>
    </Container>
  );
};

ScreenContainer.propTypes = {
  children: node.isRequired
};

export default ScreenContainer;
