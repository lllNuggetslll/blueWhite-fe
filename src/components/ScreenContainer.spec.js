import React from "react";
import ScreenContainer from "./ScreenContainer";
import { shallow } from "enzyme";

describe("Screen Container", () => {
  const child = <div />;
  const wrapper = shallow(<ScreenContainer>{child}</ScreenContainer>);

  it("should render", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
