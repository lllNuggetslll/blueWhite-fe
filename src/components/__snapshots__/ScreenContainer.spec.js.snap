// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Screen Container should render 1`] = `
<styled.div>
  <Connect(withRouter(TopBar)) />
  <styled.div>
    <div />
  </styled.div>
</styled.div>
`;
